File: documentation/RosCurve.R
library(cffdrs)
#~ library(data.table)
#~ library(dplyr)
#~ data("test_fbp")
#~ df <- test_fbp

df <- data.frame(FFMC=seq(0, 1010) / 10)
df[,"FuelType"] <- "C2"
df[,"LAT"] <- 48.970082
df[,"LONG"] <- -93.7019721
df[,"ELV"] <- 100
df[,"BUI"] <- 70
df[,"WS"] <- 20.0
df[,"WD"] <- 0
df[,"GS"] <- 50
df[,"Dj"] <- 180
df[,"D0"] <- NA
df[,"hr"] <- 1.0
df[,"PC"] <- NA
df[,"PDF"] <- NA
df[,"GFL"] <- NA
df[,"cc"] <- NA
df[,"theta"] <- 0
df[,"Accel"] <- 0
df[,"Aspect"] <- 0


f <- sort(fbp(df)[,"ROS"])
r = data.frame(FFMC=sort(df[,"FFMC"]), ROS=f)
#~ r <- merge(df, f, all.x=TRUE)[,c("FFMC", "ROS")]
write.table(r, "C:/FireGUARD/documentation/ros.csv", sep=",", row.names=FALSE)

#~ ffmc <- read.csv('C:/Users/evensjo/OneDrive - Government of Ontario/work/fban/testFARduration/ffmc_pm.csv')

#~ ffmcs <- t(ffmc[ffmc[,"LST"] %in% c(1500, 1700),])

#~ peak <- merge(ffmc, r, by.x=c("X1700"), by.y=c("FFMC"), all.x=TRUE)[,c("FFMC", "X1700", "X1500", "ROS")]

#~ at_1500 <- merge(ffmc, r, by.x=c("X1500"), by.y=c("FFMC"), allow.cartesian=TRUE)

#~ j <- merge(peak, r, by.x=c("X1500"), by.y=c("FFMC"), all.x=TRUE)

File: documentation/RosCurve.py
import pandas as pd
import math


ros = pd.read_csv('C:/FireGUARD/documentation/ros.csv')[['FFMC', 'ROS']]
ffmc = pd.read_csv('C:/Users/evensjo/OneDrive - Government of Ontario/work/fban/testFARduration/ffmc_pm.csv').set_index(['LST']).transpose()[[1500, 1700]]


r = pd.merge(ffmc, ros, left_on=1700, right_on='FFMC', how='left')[[1500, 1700, 'ROS']]
r.columns = [1500, 1700, 'ROS_1700']

s = pd.merge(r, ros, left_on=1500, right_on='FFMC', how='left')[[1500, 1700, 'ROS_1700', 'ROS']]

s.columns = [1500, 1700, 'ROS_1700', 'ROS_1500']

def find_p(x):
    return 1 / (1 + math.exp(1.64 - 0.16 * x))

s['P_1700'] = s.apply(lambda x: find_p(x['ROS_1700']), axis=1)
s['P_1500'] = s.apply(lambda x: find_p(x['ROS_1500']), axis=1)
s['pct'] = s.apply(lambda x: x['P_1500'] / x['P_1700'], axis=1)

